src/plugins/intel_gna/src/transformations/pwl/configuration_tanh.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "configuration_tanh.hpp"
+
+#include "common.hpp"
+#include "ie_common.h"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+namespace pwl {
+
+const std::string ConfigurationTanh::kActivationName = "Tanh";
+constexpr const double ConfigurationTanh::kLowerBound;
+constexpr const double ConfigurationTanh::kUpperBound;
+constexpr const double ConfigurationTanh::kMinValue;
+constexpr const double ConfigurationTanh::kMaxValue;
+constexpr const double ConfigurationTanh::kPerformanceErrorPercentage;
+constexpr const double ConfigurationTanh::kAccuracyErrorPercentage;
+constexpr const double ConfigurationTanh::kBreakBound;
+constexpr const size_t ConfigurationTanh::kMaxIterations;
+
+double ConfigurationTanh::FunctionTanh::get_value(double x) const {
+    return tanh(x);
+}
+
+double ConfigurationTanh::FunctionTanh::get_first_derivative(double x) const {
+    return 1.0 - tanh(x) * tanh(x);
+}
+
+std::shared_ptr<ActivationRepresentationConfiguration> ConfigurationTanh::create_config(PWLApproximationMode mode) {
+    return std::make_shared<ConfigurationTanh>(mode, std::make_shared<BoundariesFQHandlerImpl>());
+}
+
+ConfigurationTanh::ConfigurationTanh(PWLApproximationMode mode, std::shared_ptr<BoundariesFQHandler> boundaries_handler)
+    : ActivationRepresentationConfigurationBase(std::move(boundaries_handler)),
```

Comment:
are you going to avoid passing param through reference, that's why move semantics?

If yes, then not convinced if taking control over rvalue parameter passing is desirable here.
will copy elision handle it on it's own? - you are passing rvalue at the beginning of call chain. If not in C++11, then in C++17?... but we won't fix code when we'll be in C++17.

Decision factors:
1. is this code running under performance path? IMO not - it's at configuration phase.
2. Code readability/simplicity/maintainability might also be taken into account.
3. is similar solution observable in other OV parts?

Reply:
I want to have copy of shared_ptr - not copy of reference.
In such case it does not matter if I pass by const reference and assign to new object or I  pass by value and move.
Event more second option is better in case you create rvalue( create share_ptr as argument during invocation of method).

Reply:
>In such case it does not matter if I pass by const reference and assign to new object or I pass by value and move.
copy elision is such a compiler intelligence that for rvalues, it eliminates all the move constructor calls on the way. it does not require 'std::move' for it  - you just depend on compiler - code is cleaner, you don't justify compiler. The question is if it'll be enabled here (ie prerequisites satisfied). I'll try to check it during the day.